components/connector/internal/api/token_resolver.go

```diff
@@ -28,28 +28,28 @@ func NewTokenResolver(tokenService tokens.Service) TokenResolver {
 }
 
 func (r *tokenResolver) GenerateApplicationToken(ctx context.Context, appID string) (*externalschema.Token, error) {
-	r.log.Infof("Generating token for %s Application...", appID)
+	r.log.Infof("Generating token for Application with id %s", appID)
 
 	token, err := r.tokenService.CreateToken(appID, tokens.ApplicationToken)
 	if err != nil {
-		r.log.Error(err.Error())
+		r.log.Errorf("Error occurred while creating token for Application with id %s : %s ", appID, err.Error())
```

Comment:
i didnt see an error presenter if connector so if it works fine for director error logging i think adding one here as well would make sure sense instead of logging every error separately ? Also it's worth checking if the currently returned errors contain the additional details you added in the log entry and if not you can also enrich the returned errors whereever it seems necessary